File: Practica_11/src/Java/Fila_A/Main.java
package Java.Fila_A;

public class Main {
	public static void main(String[] args) {
		//a) Crear un objeto pintura que tenga un anuncio y otro objeto pintura sin anuncio 
		Artista a1 = new Artista("Franco", 2468, 4);
		Artista a2 = new Artista("Jael", 1357, 6);
		Artista a3 = new Artista("Jesika", 7531, 3);
		Artista a4 = new Artista("Emanuel", 8642, 8);
		
		Anuncio an1 = new Anuncio(1, 2400);
		
		Pintura p1 = new Pintura("Monalisa", "Oleo", a1, a2, an1, "Contemporaneo");
		Pintura p2 = new Pintura("Titanic", "Tela", a3, a4, null, "Paisaje");
		//Mostramos los Objetos
		p1.mostrar();
		
		//b) Mostrar el nombre del artista con más años de Experiencia de ambas pinturas
		System.out.println("\tINCISO B");
		p1.MostrarNombreArtistaMasExperiencia();
		p2.MostrarNombreArtistaMasExperiencia();
		
		//c) Se ha decidido vender la pintura sin anuncio, se pide agregar un anuncio de venta y determine el
		//monto total de venta de ambas pinturas. 
		//Creamos el Anuncio y Agregamos a la Pintura 2
		System.out.println("\tINCISO C");
		Anuncio an2 = new Anuncio(1, 1500);
		p2.b=an2;
		p2.mostrar();
		//Sumamos el Precio total de la Pintura 1 y Pintura 2
		System.out.println("MONTO TOTAL DE VENTA DE LA PINTURA 1 Y PINTURA 2");
		double precio_total = p1.b.getPrecio()+p2.b.getPrecio();
		System.out.println("El Precio es: "+precio_total);
	}
}

File: Practica_11/Python/Fila_A/Main.py
from Anuncio import Anuncio
from Artista import Artista
from Pintura import Pintura
class Main:
    @staticmethod
    def main():
        # a) Crear un objeto pintura que tenga un anuncio y otro sin anuncio
        a1 = Artista("Franco", 2468, 4)
        a2 = Artista("Jael", 1357, 6)
        a3 = Artista("Jesika", 7531, 3)
        a4 = Artista("Emanuel", 8642, 8)
        an1 = Anuncio(1, 2400)
        p1 = Pintura("Monalisa", "Oleo", a1, a2, an1, "Contemporaneo")
        p2 = Pintura("Titanic", "Tela", a3, a4, None, "Paisaje")
        # Mostramos los objetos
        p1.mostrar()
        # b) Mostrar el nombre del artista con más años de experiencia
        print("\tINCISO B")
        p1.mostrar_nombre_artista_mas_experiencia()
        p2.mostrar_nombre_artista_mas_experiencia()
        # c) Agregar un anuncio de venta a la pintura sin anuncio
        print("\tINCISO C")
        an2 = Anuncio(2, 1500)
        p2.b = an2
        p2.mostrar()
        # Calcular el monto total de venta
        print("MONTO TOTAL DE VENTA DE LA PINTURA 1 Y PINTURA 2")
        precio_total = 0
        if p1.b is not None:
            precio_total += p1.b.getPrecio()
        if p2.b is not None:
            precio_total += p2.b.getPrecio()
        print("El Precio es:", precio_total)
# Ejecutamos
if __name__ == "__main__":
    Main.main()
